src/row-builder.js

```diff
@@ -72,7 +71,7 @@ RowBuilder.formatValue = function(field, value) {
   }
 
   if (field.code !== 'STRUCT') {
-    return value;
+    return codec.decode(value, field);
```

Comment:
Here's where we let codec.decode convert the raw values to our custom values, e.g. `codec.Float`.